### PR TITLE
Support 1-4 configurable fields with dynamic font sizes

### PR DESCRIPTION
**Pull request by villekoskelaorg**
Adding support for 1-4 configurable fields.
Adding support for dynamic font size depending on amount of active fields.
Memory consumption optimizations.

### CODE REVIEW

File: source/RunScribeDataField.mc [hunk 1]
```diff
@@ -94,14 +96,15 @@ class RunScribeDataField extends Ui.DataField {
             mMetricCount = 4; 
         } else if (mMetric3Type != 0) {
             mMetricCount = 3;
-        } else {
+            mMetric4Type = mMetric3Type;
```
Comment (vparashchak) [inline source/RunScribeDataField.mc]: Please, remove this in the next commit/pull-request.

File: source/RunScribeDataField.mc [hunk 2]
```diff
@@ -94,14 +96,15 @@ class RunScribeDataField extends Ui.DataField {
             mMetricCount = 4; 
         } else if (mMetric3Type != 0) {
             mMetricCount = 3;
-        } else {
+            mMetric4Type = mMetric3Type;
+            mMetric4Type = 0;
```
Comment (vparashchak) [inline source/RunScribeDataField.mc]: And this as well.

File: source/RunScribeDataField.mc [hunk 3]
```diff
@@ -94,14 +96,15 @@ class RunScribeDataField extends Ui.DataField {
...
             mMetricCount = 2;
+        } else {
+            mMetricCount = 1;
         }
         
-        mMetric1Name = getMetricName(mMetric1Type);
-        mMetric2Name = getMetricName(mMetric2Type);
-        mMetric3Name = getMetricName(mMetric3Type);
-        mMetric4Name = getMetricName(mMetric4Type);
+        mUpdateLayout = true;
```
Comment (vparashchak) [inline source/RunScribeDataField.mc]: Please, assign numbers to this variable.

File: source/RunScribeDataField.mc [hunk 4]
```diff
@@ -151,24 +167,24 @@ class RunScribeDataField extends Ui.DataField {
...
+            return "Impact Gs"; // Ui.loadResource(Rez.Strings.impact_gs_app_label);
         } else if (metricType == 2) {
-            return "Braking Gs";
+            return "Braking Gs"; // Ui.loadResource(Rez.Strings.braking_gs_app_label);
         } else if (metricType == 3) {
-            return "Footstrike";
+            return "Footstrike"; // Ui.loadResource(Rez.Strings.fs_type_app_label);
         } else if (metricType == 4) {
-            return "Pronation";
+            return "Pronation"; // Ui.loadResource(Rez.Strings.pronation_app_label);
```
Comment (vparashchak) [inline source/RunScribeDataField.mc]: Please, remove comments here, above and below + add units for Pronation.

File: source/RunScribeDataField.mc [hunk 5]
```diff
@@ -203,25 +220,41 @@ class RunScribeDataField extends Ui.DataField {
...
         dc.setColor(fgColor, Gfx.COLOR_TRANSPARENT);
         
+        if (mUpdateLayout != 0) {
+            onLayout(dc);
+        }
+        
         // Update status
-        if (mSensorLeft != null && mSensorRight != null && (mSensorRight.searching == 0 || mSensorLeft.searching == 0)) {
-            var met1x, met1y, met2x, met2y, met3x = 0, met3y = 0, met4x = 0, met4y = 0;
+        if (mSensorLeft != null && mSensorRight != null && mSensorRight.data != null && mSensorLeft.data != null) {
```
Comment (vparashchak) [inline source/RunScribeDataField.mc]: We must check "searching" property here to be able to display "Searching..." when both channels are scanning. Please, revert back this line. Of course, you can add the check for "data" property as well.

File: source/RunScribeDataField.mc [hunk 6]
```diff
@@ -249,37 +283,45 @@ class RunScribeDataField extends Ui.DataField {
...
+                drawMetricOffset(dc, met4x, met4y, mMetric4Type);
+            } 
         } else {
-            var message;
-            if (mSensorLeft == null || mSensorRight == null) {
-                message = "No Channel!";
-            } else {
-                message = "Searching...";
+            var message = "No Channel!";
+            if (mSensorRight != null && mSensorLeft != null) {
```
Comment (vparashchak) [inline source/RunScribeDataField.mc]: We prefer displaying just "Searching..." here. Please, revert back the change.

**Review by vparashchak — COMMENTED** [review]
There are a few things that should be fixed before merging. Please, fix them and let me know. Thanks!

File: source/RunScribeDataField.mc (hunk 1 above)
Comment (villekoskelaorg) [reply source/RunScribeDataField.mc]: Copy paste mistake

File: source/RunScribeDataField.mc (hunk 2 above)
Comment (villekoskelaorg) [reply source/RunScribeDataField.mc]: Same as above

File: source/RunScribeDataField.mc (hunk 4 above)
Comment (villekoskelaorg) [reply source/RunScribeDataField.mc]: No space for units on all screen sizes.

File: source/RunScribeDataField.mc (hunk 5 above)
Comment (villekoskelaorg) [reply source/RunScribeDataField.mc]: data != null check is enough

Comment (vparashchak) [reply source/RunScribeDataField.mc]: But you never set null to this property, so if the watch loses connectivity there is no way for displaying "Searching...".

Comment (villekoskelaorg) [reply source/RunScribeDataField.mc]: Searching is handled in the later else branch.

Comment (vparashchak) [reply source/RunScribeDataField.mc]: For this we use "searching" property as is set to true when we re-open channel.

Comment (villekoskelaorg) [reply source/RunScribeDataField.mc]: Ok yes, lost connection does not cause "searching" to be displayed, only initial search is visible. If searching can occur later then this should check against "searching".

Comment (vparashchak) [reply source/RunScribeDataField.mc]: yes, that's correct.  I was talking to Tim yesterday about this and we decided that it's better to show "Searching..." when it occurs.

**Pull request merged**